File: src/main/java/kavi/tech/service/service/UserService.java
package kavi.tech.service.service;

import io.vertx.core.json.JsonArray;
import kavi.tech.service.mybatis.mapper.UserMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import rx.Single;
import tech.kavi.vs.mybatis.AsyncHelperKt;
@Service
public class UserService {
    @Autowired
    private UserMapper userMapper;

    /**
     * 查询所有用户结果
     * */
    public Single<JsonArray> listAll() {
        return AsyncHelperKt.singleAsync(userMapper::list).map(JsonArray::new);
    }
}

File: src/main/kotlin/kavi/tech/service/ms/UserService.kt
package kavi.tech.service.ms

import io.vertx.codegen.annotations.GenIgnore
import io.vertx.codegen.annotations.ProxyGen
import io.vertx.codegen.annotations.VertxGen
import io.vertx.core.AsyncResult
import io.vertx.core.Handler
import io.vertx.core.Vertx
import io.vertx.core.json.JsonArray
import io.vertx.core.json.JsonObject
import io.vertx.serviceproxy.ServiceProxyBuilder




/**
 * ProxyGen 声明要处理的Java接口类型，以生成可通过Vert.x事件总线连接到原始API的Java代理。
 * VertxGen 声明要处理的Java接口类型，以便用polyglot Vert.x的各种语言创建API。
 * */
@ProxyGen
@VertxGen
interface UserService {

    fun list(document: JsonObject, handler: Handler<AsyncResult<JsonArray>>)

    /**
     * 注册信息
     * */
    @GenIgnore
    object Factory{
        const val name = "UserService"
        const val address = "kavi.tech.service.ms.user"

        @JvmStatic
        fun create(vertx: Vertx): UserService {
            return ServiceProxyBuilder(vertx).setAddress(address).build(UserService::class.java)
        }
    }

}